mlflow/entities/evaluation.py

```diff
@@ -0,0 +1,192 @@
+        """The evaluation ID."""
+        return self._evaluation_id
+
+    @property
+    def run_id(self) -> str:
+        """The ID of the MLflow Run containing the evaluation"""
+        return self._run_id
+
+    @property
+    def inputs_id(self) -> str:
+        """The evaluation inputs ID."""
+        return self._inputs_id
+
+    @property
+    def inputs(self) -> Dict[str, Any]:
+        """The evaluation inputs."""
+        return self._inputs
+
+    @property
+    def outputs(self) -> Optional[Dict[str, Any]]:
+        """The evaluation outputs."""
+        return self._outputs
+
+    @property
+    def request_id(self) -> Optional[str]:
+        """The evaluation request ID."""
+        return self._request_id
+
+    @property
+    def targets(self) -> Optional[Dict[str, Any]]:
+        """The evaluation targets."""
+        return self._targets
+
+    @property
+    def error_code(self) -> Optional[str]:
+        """The evaluation error code."""
+        return self._error_code
+
+    @property
+    def error_message(self) -> Optional[str]:
+        """The evaluation error message."""
+        return self._error_message
+
+    @property
+    def assessments(self) -> Optional[List[Assessment]]:
+        """The evaluation assessments."""
+        return self._assessments
+
+    @property
+    def metrics(self) -> Optional[List[Metric]]:
+        """The evaluation metrics."""
+        return self._metrics
+
+    @property
+    def tags(self) -> Optional[List[EvaluationTag]]:
+        """The evaluation tags."""
+        return self._tags
+
+    def __eq__(self, __o):
+        if isinstance(__o, self.__class__):
+            return self.to_dictionary() == __o.to_dictionary()
+        return False
+
+    def to_dictionary(self) -> Dict[str, Any]:
+        """
+        Convert the Evaluation object to a dictionary.
+
+        Returns:
+            dict: The Evaluation object represented as a dictionary.
+        """
+        evaluation_dict = {
+            "evaluation_id": self.evaluation_id,
+            "run_id": self.run_id,
+            "inputs_id": self.inputs_id,
+            "inputs": self.inputs,
+        }
+        if self.outputs:
+            evaluation_dict["outputs"] = self.outputs
+        if self.request_id:
+            evaluation_dict["request_id"] = self.request_id
+        if self.targets:
+            evaluation_dict["targets"] = self.targets
+        if self.error_code:
+            evaluation_dict["error_code"] = self.error_code
+        if self.error_message:
+            evaluation_dict["error_message"] = self.error_message
+        if self.assessments:
+            evaluation_dict["assessments"] = [assess.to_dictionary() for assess in self.assessments]
+        if self.metrics:
+            evaluation_dict["metrics"] = [metric.to_dictionary() for metric in self.metrics]
+        if self.tags:
+            evaluation_dict["tags"] = [tag.to_dictionary() for tag in self.tags]
+        return evaluation_dict
```

Comment:
WDYT about a comprehension to break up the if statements?

```python
def to_dictionary(self) -> Dict[str, Any]:
        """
        Convert the Evaluation object to a dictionary.
        Returns:
            dict: The Evaluation object represented as a dictionary.
        """
        evaluation_dict = {
            "evaluation_id": self.evaluation_id,
            "run_id": self.run_id,
            "inputs_id": self.inputs_id,
            "inputs": self.inputs,
            "outputs": self.outputs,
            "request_id": self.request_id,
            "targets": self.targets,
            "error_code": self.error_code,
            "error_message": self.error_message,
            "assessments": [assess.to_dictionary() for assess in self.assessments] if self.assessments else None,
            "metrics": [metric.to_dictionary() for metric in self.metrics] if self.metrics else None,
            "tags": [tag.to_dictionary() for tag in self.tags] if self.tags else None,
        }
        return {k: v for k, v in evaluation_dict.items() if v is not None}
```

Reply:
Great idea - applied this suggestion!